src/pack/ekat_pack.hpp

```diff
@@ -620,6 +619,20 @@ operator << (std::ostream& os, const PackType& p) {
 
 } // namespace ekat
 
+// Kokkos-compatible reduction identity for arbitrary packs
+namespace Kokkos {
+template<typename S, int N>
+struct reduction_identity<ekat::Pack<S,N>> {
```

Comment:
This was moved from kokkos utils, since it belongs here, next to the type we're defining.